bondcpp/src/bond.cpp

```diff
@@ -412,19 +401,25 @@ bool Bond::waitUntilFormed(rclcpp::Duration timeout)
       break;  //  The deadline has expired
     }
     r.sleep();
+
+    std::unique_lock<std::mutex> lock(state_machine_mutex_);
+    if (sm_.getState().getId() != SM::WaitingForSister.getId()) {
+      formed = true;
+      break;
+    }
   }
 
-  return sm_.getState().getId() != SM::WaitingForSister.getId();
+  return formed;
 }
 
 bool Bond::waitUntilBroken(rclcpp::Duration timeout)
 {
-  //  std::unique_lock<std::mutex> lock(mutex_);
   rclcpp::Clock steady_clock(RCL_STEADY_TIME);
```

Comment:
It looks like we are instantiating the same variable here and then down in the `while (!broken)` link, creating a shadow.  We probably only need one of these.

Reply:
Removed the definition in the loop scope.